docs/dev/release-process.md

```diff
@@ -0,0 +1,26 @@
+# The Release Process
+
+This project currently follows a [Continuous Delivery](https://en.wikipedia.org/wiki/Continuous_delivery) process, but it's gradually moving toward [Continuous Deployment](https://en.wikipedia.org/wiki/Continuous_deployment).
+
+Whenever a commit is merged to this repository's `main` branch, the deployment pipeline kicks in, deploying the changeset to the [`stage` environment](../firefox.md#stage).
```

Comment:
I believe commit is the right word here, because the deployment pipeline kicks in even if you directly commit to the main branch without a the commit coming from another branch. Or did you mean something else, @hackebrot ?

Reply:
That makes sense. I would rephrase to

> Whenever a commit is pushed to this repository's `main` branch.

in that case, because you cannot "merge" a single Git commit.